core/unit_test/TestReducers.hpp

```diff
@@ -1020,7 +1020,7 @@ struct TestReducers {
 
   static void execute_integer() {
     test_sum(10001);
-    test_prod(35);
+    test_prod(sizeof(Scalar) > 4 ? 35 : 19);  // avoid int overflow
```

Comment:
I would rather branch on the largest representable number such that
```
std::numeric_limits<Scalar>::max() > std::pow(4,n)
n < std::log(std::numeric_limits<Scalar>::max())/std::log(4);
```

Reply:
I can expand on the comment but your suggestion is imperfect too because it use knowledge of the fact `test_prod` generates random numbers between 1 and 4.

Reply:
For the record `std::log(std::numeric_limits<int>::max())/std::log(4)` is 15.5

Reply:
Yes, so why is 19 guaranteed to work?